plugins/modules/system/alternatives.py

```diff
@@ -70,6 +87,15 @@
 from ansible.module_utils.basic import AnsibleModule
 
 
+class AlternativeState:
```

Comment:
A bit of a nit-picking but I think both terms in the name should be plural:

- the name of the module is "alternatives" so we should be consistent
- within this class we have more than one state, so it should be "states". Line 109 `choices=AlternativeState.to_list()` sounds quite odd - to me at least.

Also I think we might want to consider prefixing that class name with a `_`, so that it doesn't become a public symbol. By being public, any change we make later on will require a long deprecation period.

Reply:
There are many arguments for and against using a plural vs. a singular name for an enumeration. Java and C# seem to prefer singular names (https://stackoverflow.com/a/15756009). I guess in the end it's a matter of taste.

> Also I think we might want to consider prefixing that class name with a `_`, so that it doesn't become a public symbol. By being public, any change we make later on will require a long deprecation period.

I wouldn't consider modules (as opposed to module_utils etc.) a public API. The [guidelines only talk about public/private module and plugin utils](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#standards-for-developing-module-and-plugin-utilities). Maybe we should discuss this in community-topics...

Reply:
Fair enough. I reckon this is not a show stopper.